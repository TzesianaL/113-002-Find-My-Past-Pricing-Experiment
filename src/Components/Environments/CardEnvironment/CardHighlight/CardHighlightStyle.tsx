import styled from 'styled-components'

export const CardHighlightStyle = styled.div<{ isSelected?: boolean }>`
  border-color: ${(props) => (props.isSelected ? '#00b050' : '#00460f')};
  background-color: ${(props) => (props.isSelected ? '#00b050' : '#00460f')};
  color: white;
  border-top-left-radius: ${(props) =>
    props.theme.card?.borderRadius || '0.3em'};
  border-top-right-radius: ${(props) =>
    props.theme.card?.borderRadius || '0.3em'};
  position: absolute;
  // z-index: -1;
  text-align: left;
  width: 55%;
  /* width: ${(props) => props.theme.card?.width || '180px'}; */
  min-height: ${(props) => props.theme.highlight?.height || '0.5em'};
  padding: 0.5em 0 0.25em 0.7em;
  display: flex;
  flex-direction: column;
  top: -2em;
  left: -0.1em;
`
