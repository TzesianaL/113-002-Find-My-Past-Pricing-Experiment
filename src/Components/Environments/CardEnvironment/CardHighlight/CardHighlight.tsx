import React, { FC } from 'react'
import { CardHighlightStyle } from './CardHighlightStyle'

interface CardHighlightProps {
  children: React.ReactNode
  isSelected?: boolean
}

const CardHighlight: FC<CardHighlightProps> = ({
  children,
  isSelected = false,
}) => {
  return (
    <CardHighlightStyle isSelected={isSelected}>{children}</CardHighlightStyle>
  )
}

export default CardHighlight
