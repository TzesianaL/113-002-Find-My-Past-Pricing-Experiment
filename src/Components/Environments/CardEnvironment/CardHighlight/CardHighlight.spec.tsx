import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import CardHighlight from './CardHighlight'

describe('<CardHighlight />', () => {
  it('should render correctly Highlighted Card', () => {
    const tree = renderer.create(<CardHighlight />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
