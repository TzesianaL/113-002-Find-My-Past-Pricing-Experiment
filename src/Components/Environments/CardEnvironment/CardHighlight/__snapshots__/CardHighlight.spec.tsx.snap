// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHighlight /> should render correctly Highlighted Card 1`] = `
.c0 {
  border-color: #FFC000;
  background-color: yellow;
  color: #000000;
  border-top-style: solid;
  border-left-style: solid;
  border-right-style: solid;
  border: solid 2px;
  border-top-left-radius: 1.5em;
  border-top-right-radius: 1.5em;
  position: absolute;
  z-index: -1;
  text-align: center;
  width: 100%;
  min-height: 2.7em;
  padding: 0.5em 0 0.25em 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  top: -2.5em;
}

<div
  className="c0"
/>
`;
